src/lib/pipeline/m2/material/index.js

```diff
@@ -257,6 +270,58 @@ class M2Material extends THREE.ShaderMaterial {
     }
   }
 
+  registerAnimations(def) {
+    const { uvAnimations, transparencyAnimations, vertexColorAnimation } = def;
+
+    this.registerUVAnimations(uvAnimations);
+    this.registerTransparencyAnimations(transparencyAnimations);
+    this.registerVertexColorAnimation(vertexColorAnimation);
+  }
+
+  registerUVAnimations(uvAnimationIndices) {
+    const { animations, uvAnimationValues } = this.m2;
+
+    uvAnimationIndices.forEach((uvAnimationIndex, opIndex) => {
+      const target = this.uniforms.animatedUVs;
+      const source = uvAnimationValues[uvAnimationIndex];
+
+      animations.on('update', () => {
+        target.value[opIndex] = source.matrix;
+      });
```

Comment:
Oh! Hah! That's a very good point. I think I should be able to share the listeners like that. I suppose I could even share between UV, transparency, and vertex color animations, actually.

I'll try get that set up and handle removal of event listeners when the M2 is being cleaned up. I'd vote for waiting to merge this PR until I do that.